packages/dss-bridge/src/DomainHost.sol

```diff
@@ -154,6 +156,34 @@ abstract contract DomainHost {
 
     // --- MCD Support ---
 
+    /// @notice Enqueue a message to be played back in order
+    /// @param id The order id for the message
+    /// @param data The function call to execute
+    function enqueue(uint256 id, bytes calldata data) external guestOnly {
+        queue[id] = data;
+    }
+
+    /// @notice Play back the next message if available
+    function next() external {
+        uint256 _nextId = nextId + 1;
+        require(queue[_nextId].length > 0, "DomainHost/message-unavailable");
+
+        (bool success, bytes memory response) = address(this).call(queue[_nextId]);
+        if (success) {
+            nextId = _nextId;
```

Comment:
If the execution of an enqueued message happens to (permanently) revert, the queue will no longer be usable for future messages. That seems fragile. Maybe there should be a way to unblock or reset the queue.

Reply:
Yeah agreed.